### PR TITLE
Domains: Use destructured `selectedSite` prop in `TransferToOtherUser`

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

This PR addresses a minor cleanup as a follow-up to #51809, as suggested in https://github.com/Automattic/wp-calypso/pull/51809#discussion_r611535174

#### Testing instructions

* Verify `/domains/manage/:domain/transfer/other-user/:site` still works as it did before for Atomic and simple sites, specifically loading users and selecting one to transfer.

### CODE REVIEW

File: client/components/data/domain-management/index.jsx [hunk 1]
```diff
@@ -63,12 +63,12 @@ class DomainManagementData extends React.Component {
...
 					{ React.createElement( this.props.component, {
 						context: this.props.context,
-						domains: this.props.selectedSite ? this.props.domains : null,
+						domains: selectedSite ? this.props.domains : null,
 						hasSiteDomainsLoaded: this.props.hasSiteDomainsLoaded,
 						isRequestingSiteDomains: this.props.isRequestingSiteDomains,
 						products: this.props.products,
 						selectedDomainName: this.props.selectedDomainName,
-						selectedSite: this.props.selectedSite,
+						selectedSite: selectedSite,
```
Comment (flootr) [inline client/components/data/domain-management/index.jsx]: nit: just my personal preference, but we could shorten this to just `selectedSite,` – also, is there a reason we only destructure certain props and not all? don't feel obligued to change this, it's purely stylistic.

**Review by flootr — APPROVED** [review]
👍🏻

File: client/components/data/domain-management/index.jsx (hunk 1 above)
Comment (tyxla) [reply client/components/data/domain-management/index.jsx]: > nit: just my personal preference, but we could shorten this to just selectedSite,

Done, good call!

> also, is there a reason we only destructure certain props and not all? don't feel obligued to change this, it's purely stylistic.

No reason 😉  Just trying to limit the number of changes I guess.

**Review by jsnajdr — APPROVED** [review]
Thanks!
